app/src/main/java/mozilla/lockbox/presenter/ItemDetailPresenter.kt

```diff
@@ -94,7 +95,14 @@ class ItemDetailPresenter(
             .filterNotNull()
             .doOnNext { credentials = it }
             .map { it.toDetailViewModel() }
-            .subscribe(view::updateItem)
+            .subscribe {
+                if (it.username.isNullOrEmpty()) {
+                    view.showUsernamePlaceholder()
+                } else {
+                    view.showUsername(it.username)
+                    view.updateItem(it)
```

Comment:
what happens when there is a null username, but there is a password / hostname? do the password and hostname not get updated?